src/IIIFPresentation/API/Features/Manifest/CanvasPaintingResolver.cs

```diff
@@ -235,6 +258,21 @@ string GetCanvasId(Models.API.Manifest.CanvasPainting? cp)
         }
     }
 
+    private string? GetCanvasId(Models.API.Manifest.CanvasPainting? canvasPainting, int customerId)
```

Comment:
This seems suited to `PathParser` class? Keeps all path-parsing utils together and makes it easier to test etc